controllers/crd/crd_controller.go

```diff
@@ -0,0 +1,73 @@
+package crd
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/klog/v2"
+	"reflect"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// CustomResourceDefinitionReconciler reconciles a CustomResourceDefinition object
+// nolint:revive
+type CustomResourceDefinitionReconciler struct {
+	Client        client.Client
+	ctx           context.Context
+	Log           logr.Logger
+	AvailableCrds map[string]bool
+}
+
+// Reconcile compares available CRDs maps following either a Create or Delete event
+func (r *CustomResourceDefinitionReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	r.ctx = ctx
+	r.Log.Info("Reconciling CustomResourceDefinition.", "CRD", klog.KRef(request.Namespace, request.Name))
+
+	var err error
+	availableCrds, err := util.MapCRDAvailability(ctx, r.Client, util.CRDList...)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if !reflect.DeepEqual(availableCrds, r.AvailableCrds) {
```

Comment:
I do not believe a map guarantees the order of keys if the 2 maps contain different keys. DeepEqual might fail even if both maps contain the exact same keys.  

Please move to a "len check + loop on items" for checking equality.